### PR TITLE
Add big-list-of-naughty-strings tests

### LINKED ISSUE #151: Should test against BLNS
The Big List of Naughty Strings got some attention this week: https://github.com/minimaxir/big-list-of-naughty-strings

They have a nice JSON download. I'd like to start including them as sanity checks for the parser. Just make sure it doesn't crash or infinite-loop or something.

### PR DESCRIPTION
**Pull request by srawlins**
Fixes #151 

Kind of a weird testing feature... I was hoping that it would find some flaw in the package, but the tests pass as-is quickly, without issue. 🤷‍♂️  meh.

### CODE REVIEW

File: test/blns.dart [hunk 1]
```diff
@@ -0,0 +1,515 @@
+// GENERATED FILE. DO NOT EDIT.
+//
+// This file was generated from big-list-of-naughty-strings's JSON file:
+// https://github.com/minimaxir/big-list-of-naughty-strings/raw/master/blns.json
+// at 2017-12-28 09:31:32.140717 by the script, tool/update_blns.dart.
+
+var blns = <String>[
```
Comment (kevmoo) [inline test/blns.dart]: const?

File: tool/update_blns.dart [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import 'dart:convert';
+import 'dart:io';
+
+String _blnsJsonRawUrl =
```
Comment (kevmoo) [inline tool/update_blns.dart]: final both of these? Don't need the type...

**Comment by kevmoo** [comment]
@srawlins – thoughts on this?

**Comment by srawlins** [comment]
I was hesitant to land this as it adds 1000 test cases, but it also apparently only adds 2 seconds to run the tests so... probably fine!

File: test/blns.dart (hunk 1 above)
Comment (srawlins) [reply test/blns.dart]: Done.

File: tool/update_blns.dart (hunk 2 above)
Comment (srawlins) [reply tool/update_blns.dart]: Done.

File: tool/update_blns.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import 'dart:convert';
+import 'dart:io';
+
+final _blnsJsonRawUrl =
+    'https://github.com/minimaxir/big-list-of-naughty-strings/raw/master/blns.json';
+final _blnsFilePath = 'test/blns.dart';
+
+Future<Null> main() async {
```
Comment (kevmoo) [inline tool/update_blns.dart]: Wonder if it'd be better to just store the JSON file and read from that instead.

Then you don't have to worry that you encoded everything correctly.

File: tool/update_blns.dart [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  for (var str in json) {
+    var escaped = str
+        .replaceAll(r'\', r'\\')
+        .replaceAll("'", r"\'")
+        .replaceAll(r'$', r'\$');
+    blnsContent.writeln("  '$escaped',");
+  }
+  blnsContent.writeln('];');
+  new File(_blnsFilePath)..writeAsStringSync(blnsContent.toString());
+  exit(0);
```
Comment (kevmoo) [inline tool/update_blns.dart]: Seems superfluous

Comment (srawlins) [reply tool/update_blns.dart]: It hangs without this...

**Comment by kevmoo** [comment]
Close/flush the writer?

File: tool/update_blns.dart (hunk 4 above)
Comment (srawlins) [reply tool/update_blns.dart]: I've tried a few methods. Nothing exits faster than like, 10 seconds, without exit:

```dart
new File(_blnsFilePath)..writeAsStringSync(blnsContent.toString(), flush:true);
```

```dart
(new File(_blnsFilePath).openWrite())..write(blnsContent.toString())..close();
```

Any other suggestions?

**Comment by kevmoo** [comment]
Ah! T'was the HTTPClient

```diff
diff --git a/tool/update_blns.dart b/tool/update_blns.dart
index 7fd7bb6..c5a2784 100644
--- a/tool/update_blns.dart
+++ b/tool/update_blns.dart
@@ -8,10 +8,17 @@ final _blnsFilePath = 'test/blns.dart';

 Future<Null> main() async {
   var client = new HttpClient();
-  var request = await client.getUrl(Uri.parse(_blnsJsonRawUrl));
-  var response = await request.close();
-  var json = JSON.decode(await response.transform(UTF8.decoder).join(''))
-      as List<String>;
+
+  List<String> json;
+  try {
+    var request = await client.getUrl(Uri.parse(_blnsJsonRawUrl));
+    var response = await request.close();
+    json = JSON.decode(await response.transform(UTF8.decoder).join(''))
+        as List<String>;
+  } finally {
+    client.close();
+  }
+
   var blnsContent = new StringBuffer('''
 // GENERATED FILE. DO NOT EDIT.
 //
@@ -30,5 +37,4 @@ Future<Null> main() async {
   }
   blnsContent.writeln('];');
   new File(_blnsFilePath)..writeAsStringSync(blnsContent.toString());
-  exit(0);
 }
```

**Comment by srawlins** [comment]
Woohoo that did it! Thanks @kevmoo 🎉 🎉 🎉 !!

File: tool/update_blns.dart (hunk 3 above)
Comment (srawlins) [reply tool/update_blns.dart]: I'll leave this for now.

Comment (kevmoo) [reply tool/update_blns.dart]: ack

**Pull request merged**